src/frequenz/sdk/microgrid/component_data.py

```diff
@@ -17,229 +19,218 @@
+    active_power: float
+    current_per_phase: Tuple[float, float, float]
+    voltage_per_phase: Tuple[float, float, float]
 
-        Returns:
-            Current values for each phase.
-        """
-        phase_1 = self._raw.meter.data.ac.phase_1.current.value
-        phase_2 = self._raw.meter.data.ac.phase_2.current.value
-        phase_3 = self._raw.meter.data.ac.phase_3.current.value
-        return (phase_1, phase_2, phase_3)
+    @staticmethod
+    def from_proto(raw: microgrid_pb.ComponentData) -> MeterData:
+        """Create MeterData from a protobuf message.
 
-    @property
-    def voltage_per_phase(self) -> Tuple[float, float, float]:
-        """Get voltages measured by the meter per phase.
+        Args:
+            raw: protobuf message
 
         Returns:
-            Voltage values for each phase.
+            Instance of MeterData created from the protobuf message.
         """
-        phase_1 = self._raw.meter.data.ac.phase_1.voltage.value
-        phase_2 = self._raw.meter.data.ac.phase_2.voltage.value
-        phase_3 = self._raw.meter.data.ac.phase_3.voltage.value
-        return (phase_1, phase_2, phase_3)
+        return MeterData(
+            component_id=raw.id,
+            timestamp=raw.ts.ToDatetime(tzinfo=pytz.UTC),
+            active_power=raw.meter.data.ac.power_active.value,
+            current_per_phase=(
+                raw.meter.data.ac.phase_1.current.value,
+                raw.meter.data.ac.phase_2.current.value,
+                raw.meter.data.ac.phase_3.current.value,
+            ),
+            voltage_per_phase=(
+                raw.meter.data.ac.phase_1.voltage.value,
+                raw.meter.data.ac.phase_2.voltage.value,
+                raw.meter.data.ac.phase_3.voltage.value,
+            ),
+        )
 
 
-class BatteryData(_BaseComponentData):
-    """A wrapper class for holding battery data."""
+@dataclass(frozen=True)
+class BatteryData(ComponentData):
+    """A wrapper class for holding battery data.
 
-    def __init__(self, raw: microgrid_pb.ComponentData) -> None:
-        """Create a `BatteryData` instance.
+    soc: battery's overall SoC in percent (%).
 
-        Args:
-            raw: raw component data as decoded from the wire.
-        """
-        self._raw = raw
-
-    @property
-    def soc(self) -> float:
-        """Get current SoC level.
-
-        Returns:
-            SoC level.
-        """
-        return self._raw.battery.data.soc.avg
+    soc_lower_bound: the soc lower bound of the current bounds of the metric, as
+        imposed by the overall system.
 
-    @property
-    def soc_upper_bound(self) -> float:
-        """Get SoC upper bound.
+    soc_upper_bound: the soc upper bound of the current bounds of the metric, as
+        imposed by the overall system.
 
-        Returns:
-            Maximum SoC level.
-        """
-        return self._raw.battery.data.soc.system_bounds.upper
+    capacity: the capacity of the battery
```

Comment:
In which unit?

Reply:
Actually, I don't know. The comment in the protobuf message only says:

> The capacity of the battery.

Reply:
Capacity is always in `Wh` (Watt-hour)

Reply:
Added unit description in the docstring.